pkg/fakerp/config.go

```diff
@@ -55,44 +75,49 @@ func GetPluginConfig() (*api.PluginConfig, error) {
 	if err != nil {
 		return nil, err
 	}
+	template.Certificates.GenevaLogging.Cert = logCert
+	template.Certificates.GenevaLogging.Key = logKey
+	template.Certificates.GenevaMetrics.Cert = metCert
+	template.Certificates.GenevaMetrics.Key = metKey
+	template.Images.GenevaImagePullSecret = pullSecret
 
-	var syncImage, metricsBridge string
-	if os.Getenv("SYNC_IMAGE") == "" {
-		syncImage = "quay.io/openshift-on-azure/sync:latest"
-	} else {
-		syncImage = os.Getenv("SYNC_IMAGE")
+	return template, nil
+}
+
+func overridePluginTemplate(template *pluginapi.Config) {
+	if os.Getenv("SYNC_IMAGE") != "" {
+		template.Images.Sync = os.Getenv("SYNC_IMAGE")
 	}
-	if os.Getenv("METRICSBRIDGE_IMAGE") == "" {
-		metricsBridge = "quay.io/openshift-on-azure/metricsbridge:latest"
-	} else {
-		metricsBridge = os.Getenv("METRICSBRIDGE_IMAGE")
+	if os.Getenv("METRICSBRIDGE_IMAGE") != "" {
+		template.Images.MetricsBridge = os.Getenv("METRICSBRIDGE_IMAGE")
 	}
+	if os.Getenv("ETCDBACKUP_IMAGE") != "" {
+		template.Images.EtcdBackup = os.Getenv("ETCDBACKUP_IMAGE")
+	}
+	if os.Getenv("AZURE_CONTROLLERS_IMAGE") != "" {
+		template.Images.AzureControllers = os.Getenv("AZURE_CONTROLLERS_IMAGE")
+	}
+	if os.Getenv("OREG_URL") != "" {
+		template.Images.Format = os.Getenv("OREG_URL")
+	}
+	if os.Getenv("IMAGE_VERSION") != "" {
+		template.ImageVersion = os.Getenv("IMAGE_VERSION")
+	}
+	if os.Getenv("IMAGE_OFFER") != "" {
+		template.ImageOffer = os.Getenv("IMAGE_OFFER")
+	}
+}
 
-	genevaConfig := api.GenevaConfig{
-		ImagePullSecret: pullSecret,
-
-		LoggingCert:                logCert,
-		LoggingKey:                 logKey,
-		LoggingSector:              "US-Test",
-		LoggingAccount:             "ccpopenshiftdiag",
-		LoggingNamespace:           "CCPOpenShift",
-		LoggingControlPlaneAccount: "RPOpenShiftAccount",
-		LoggingImage:               "osarpint.azurecr.io/acs/mdsd:12051806",
-		TDAgentImage:               "osarpint.azurecr.io/acs/td-agent:latest",
-
-		MetricsCert:     metCert,
-		MetricsKey:      metKey,
-		MetricsBridge:   metricsBridge,
-		StatsdImage:     "osarpint.azurecr.io/acs/mdm:git-a909a2e76",
-		MetricsAccount:  "RPOpenShift",
-		MetricsEndpoint: "https://az-int.metrics.nsatc.net/",
+func readFile(path string) ([]byte, error) {
+	if fileExist(path) {
+		return ioutil.ReadFile(path)
 	}
-	return &api.PluginConfig{
-		SyncImage:       syncImage,
-		AcceptLanguages: []string{"en-us"},
-		TestConfig:      tc,
-		GenevaConfig:    genevaConfig,
-	}, nil
+	return []byte{}, fmt.Errorf("file %s does not exist", path)
+}
+
+func fileExist(path string) bool {
```

Comment:
can you revert this diff? No point in reorganizing functions when your PR is already that big. Same for `readFile`.